// Torbert, 7.20.06

   import java.awt.*;
   import java.awt.image.*;

    public class Turtle
   {
      private static BufferedImage img;
      private static int black = 0;
      private static int blue = 255;       // 2^8-1
      private static int green = 65280;    //(2^8-1)*2^8
      private static int red = 16711680;   //(2^8-1)*2^16
      private static int white = 16777215; // 2^24-1
      private double x,y,theta;            // what does a Turtle know?
      static									 	 // a static initializer list
      {
         img = new BufferedImage(800,600,BufferedImage.TYPE_INT_RGB);
      }
       public Turtle()
      {
         //img = new BufferedImage(800,600,BufferedImage.TYPE_INT_RGB);
         x=img.getWidth()/2;           //start at center
         y=img.getHeight()/2;          //start at center
         theta=90;                     //start facing north
      }
       public static Image getImage()
      {
         return img;
      }
       public void turnLeft(double a)
      {
         theta+=a;
      }
       public void forward(double r)
      {
         for(int i = 0;i < 200; i++)
         {
            x += r * Math.cos(Math.PI*(theta + 90)/180) / 200;
            y -= r * Math.sin(Math.PI*(theta + 90)/180) / 200;
            System.out.println(Math.cos(theta) + " + " +Math.sin(theta));
            img.setRGB((int)x, (int)y, white);
         }
      }
   }